dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetFieldsByType.java

```diff
@@ -0,0 +1,67 @@
+package edu.harvard.iq.dataverse.persistence.dataset;
+
+import com.google.common.base.Preconditions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Model class grouping dataset fields with the same type
+ * 
+ * @author madryk
+ */
+public class DatasetFieldsByType {
+
+    private DatasetFieldType datasetFieldType;
+
+    private List<DatasetField> datasetFields = new ArrayList<>();
+
+    private boolean include = true;
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    public DatasetFieldsByType(DatasetFieldType datasetFieldType, List<DatasetField> datasetFields) {
+        datasetFields.forEach(field -> Preconditions.checkArgument(field.getDatasetFieldType().equals(datasetFieldType)));
+
+        this.datasetFieldType = datasetFieldType;
+        this.datasetFields = datasetFields;
+    }
+
+    // -------------------- GETTERS --------------------
+
+    public DatasetFieldType getDatasetFieldType() {
+        return datasetFieldType;
+    }
+
+    public List<DatasetField> getDatasetFields() {
+        return datasetFields;
+    }
+
+    /**
+     * Returns true if fields should be visible on create/edit metadata
+     * form. If false then they should be hidden.
+     * Note that only fields of type without parent can have this setting
```

Comment:
You mean that it should be argument in a constructor and it should be validated if value could be passed?

Reply:
Yes, maybe with some kind of exception – so that one can quickly spot the failure when adding/modifying code.

Reply:
But that would mean that the class is only usable for Parent fields, since if you pass the boolean argument there would be an exception. I'm not sure if that was the intention here.

Reply:
Ps. It is Krzysiek class so I'm not sure about intention in this one.

Reply:
Ok, so let it stay as it is.